### PR TITLE
[SPARK-22540][SQL] Ensure HighlyCompressedMapStatus calculates correct avgSize

### PR DESCRIPTION
**Pull request by yucai**
## What changes were proposed in this pull request?

Ensure HighlyCompressedMapStatus calculates correct avgSize

## How was this patch tested?

New unit test added.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
ok to test

File: core/src/test/scala/org/apache/spark/scheduler/MapStatusSuite.scala [hunk 1]
```diff
@@ -98,6 +98,28 @@ class MapStatusSuite extends SparkFunSuite {
     }
   }
 
+  test("[SPARK-22540]: HighlyCompressedMapStatus's avgSize is incorrect") {
```
Comment (jiangxb1987) [inline core/src/test/scala/org/apache/spark/scheduler/MapStatusSuite.scala]: nit: `test("SPARK-22540: ensure HighlyCompressedMapStatus calculates correct avgSize")`

**Review by jiangxb1987 — APPROVED** [review]
LGTM only one nit

**Comment by srowen** [comment]
Merged to master/2.2

**Pull request closed without merging**